can/message.py

```diff
@@ -1,46 +1,115 @@
+        # can be removed in 4.0
+        # this method is only called if the attribute was not found elsewhere, like in __slots__
+        try:
+            warnings.warn("Custom attributes of messages are deprecated and will be removed in the next major version", DeprecationWarning)
+            return self._dict[key]
+        except KeyError:
+            raise AttributeError("'message' object has no attribute '{}'".format(key))
+
+    def __setattr__(self, key, value):
+        # see __getattr__
+        try:
+            super(Message, self).__setattr__(key, value)
+        except AttributeError:
+            warnings.warn("Custom attributes of messages are deprecated and will be removed in the next major version", DeprecationWarning)
+            self._dict[key] = value
+
+    @property
+    def id_type(self):
+        # TODO remove in 4.0
+        warnings.warn("Message.id_type is deprecated, use is_extended_id", DeprecationWarning)
+        return self.is_extended_id
+
+    @id_type.setter
+    def id_type(self, value):
+        # TODO remove in 4.0
+        warnings.warn("Message.id_type is deprecated, use is_extended_id", DeprecationWarning)
+        self.is_extended_id = value
+
+    def __init__(self, timestamp=0.0, arbitration_id=0, is_extended_id=None,
+                 is_remote_frame=False, is_error_frame=False, channel=None,
+                 dlc=None, data=None,
                  is_fd=False, bitrate_switch=False, error_state_indicator=False,
-                 channel=None):
+                 extended_id=True,
+                 check=False):
+        """
+        To create a message object, simply provide any of the below attributes
+        together with additional parameters as keyword arguments to the constructor.
+
+        :param bool check: By default, the constructor of this class does not strictly check the input.
+                           Thus, the caller must prevent the creation of invalid messages or
+                           set this parameter to `True`, to raise an Error on invalid inputs.
+                           Possible problems include the `dlc` field not matching the length of `data`
+                           or creating a message with both `is_remote_frame` and `is_error_frame` set to `True`.
+
+        :raises ValueError: iff `check` is set to `True` and one or more arguments were invalid
+        """
+        self._dict = dict() # see __getattr__
 
         self.timestamp = timestamp
-        self.id_type = extended_id
-        self.is_extended_id = extended_id
+        self.arbitration_id = arbitration_id
+        if is_extended_id is not None:
+            self.is_extended_id = is_extended_id
+        else:
+            if not extended_id:
+                # Passed extended_id=False (default argument is True) so we warn to update
+                warnings.warn("extended_id is a deprecated parameter, use is_extended_id", DeprecationWarning)
```

Comment:
This causes warnings only when `extended_id=False` but not on `extended_id=True`. Also we haven't updated our own usages of the old interface so developers get lots of warnings just by using the library.

Reply:
Yeah, that should be fixed first. I am currently on vacation and can only help with such things mid next week.

Reply:
Done